hugegraph-core/src/main/java/com/baidu/hugegraph/util/JsonUtil.java

```diff
@@ -44,6 +55,16 @@
         SimpleModule module = new SimpleModule();
         module.addSerializer(Date.class, new DateSerializer());
         module.addDeserializer(Date.class, new DateDeserializer());
+
+        module.addSerializer(IdGenerator.StringId.class,
+                             new IdSerializer<>(IdGenerator.StringId.class));
+        module.addSerializer(IdGenerator.LongId.class,
+                             new IdSerializer<>(IdGenerator.LongId.class));
+        module.addSerializer(EdgeId.class, new IdSerializer<>(EdgeId.class));
+
+        module.addSerializer(HugeVertex.class, new HugeVertexSerializer());
```

Comment:
To ensure that the format of writeObject()  is the same as writeIterator(), also register the serializer to GraphSONWriter.